shipshape/cli/shipshape.go

```diff
@@ -257,13 +257,19 @@ func main() {
 }
 
 func startShipshapeService(image, absRoot string, analyzers []string) (*client.Client, error) {
-	glog.Infof("Running image %s in service mode", image)
-	result := docker.RunService(image, "shipping_container", absRoot, localLogs, analyzers)
-	glog.Infoln(strings.TrimSpace(result.Stdout))
-	glog.Infoln(strings.TrimSpace(result.Stderr))
-	if result.Err != nil {
-		return nil, result.Err
+	// If this doesn't match the image, stop and restart the service.
+	// Otherwise, use the existing one.
+	if !docker.ImageMatches(image, "shipping_container") {
+		glog.Infof("Restarting container with %s", image)
+		stop("shipping_container", 0)
+		result := docker.RunService(image, "shipping_container", absRoot, localLogs, analyzers)
+		glog.Infoln(strings.TrimSpace(result.Stdout))
```

Comment:
Prefix logline with "stdout" or something to make it clear what the loglines correspond to. Ditto for the stderr.